lib/ansible/executor/task_executor.py

```diff
@@ -556,7 +556,7 @@ def _execute_internal(self, templar: TemplateEngine, variables: dict[str, t.Any]
             if not include_file:
                 return dict(failed=True, msg="No include file was specified to the include")
 
-            return dict(include=include_file, include_args=include_args)
+            return dict(include=templar.template(include_file), include_args=include_args)
```

Comment:
Manual templating on FA values should usually be the last resort- this one's right after a `task.post_validate()` call, which should leave pretty much nothing un-templated. That sure smells to me like a bug in the FA handling. The fact that [this](https://github.com/ansible/ansible/blob/03181ac87b6c62bec8ae63a896d8a986a60eae8a/lib/ansible/playbook/base.py#L570) is missing `HandlerTaskInclude` is probably the bug.

I'm not sure how this was working pre-DT (since that early bailout exclusion was also missing `TaskInclude` in 2.18), but we shuffled a few layers around for task validation to support some other DT features and make more stuff automatic. Adding `HandlerTaskInclude` there should get it to auto-template it for you in `task.post_validate`.